src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala

```diff
@@ -4,26 +4,36 @@ import org.broadinstitute.dsde.firecloud.core.GetEntitiesWithType.EntityWithType
 import org.broadinstitute.dsde.firecloud.model.ModelSchema
 
 import scala.collection.immutable
-import scala.util.Try
+import scala.util.{Success, Try}
 
 import spray.json.DefaultJsonProtocol._
 import spray.json.JsValue
 
 object TSVFormatter {
 
-  def makeMembershipTsvString(entities: List[EntityWithType], entityType: String): String = {
+  def makeMembershipTsvString(entities: List[EntityWithType], entityType: String, collectionMemberType: String): String = {
     val headers: immutable.IndexedSeq[String] = immutable.IndexedSeq("membership:" + entityType + "_id", entityType.replace("_set", "_id"))
-    val rows: List[IndexedSeq[String]] = entities filter { _.entityType == entityType } flatMap { entity =>
-      entity.attributes.getOrElse(Map.empty) flatMap { m =>
-        m._2.convertTo[List[JsValue]] map { jsValue =>
-          val cellValue: String = jsValue match {
-            case x if Try(x.asJsObject.fields.contains("entityName")).isSuccess =>
-              cleanValue(x.asJsObject.fields.getOrElse("entityName", x))
-            case _ =>
-              ""
+    val rows: List[IndexedSeq[String]] = entities.
+      filter { _.entityType == entityType }.
+      flatMap { entity =>
+        entity.attributes.getOrElse(Map.empty).
+          // To make the membership file, we need the array of elements that correspond to the set type.
+          // All other top-level properties are not necessary and are only used for the data load file.
+          filter { attribute => attribute._1.equals(collectionMemberType) }.
+          flatMap { m =>
+            Try(m._2.convertTo[List[JsValue]]) match {
+              case Success(values) =>
+                values map { jsValue =>
+                  val cellValue: String = jsValue match {
+                    case x if Try(x.asJsObject.fields.contains("entityName")).isSuccess =>
+                      cleanValue(x.asJsObject.fields.getOrElse("entityName", x))
+                    case _ =>
```

Comment:
Actually, I think I mis-spoke in my previous comment here. There are two cases when iterating over top-level attributes of an entity. In this case, this only should ever hit the `case x` case. For example, here is what we are looking at in this case:
```
"samples": [{
      "entityType": "sample",
      "entityName": "COAD-4N-A93T-NB"
    }, {
      "entityType": "sample",
      "entityName": "COAD-AA-3489-TP"
    }...
```
We should never really see the `case _` but I didn't want to throw away this field in the case the data was somehow not what is expected.